### PR TITLE
IR-91: Support Azure CloudName

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by dmage** [comment]
```
failed to wait for the imageregistry resource to be processed: Get https://api.ci-op-vh65zrbi-0ca04.ci.azure.devcluster.openshift.com:6443/apis/imageregistry.operator.openshift.io/v1/configs/cluster: read tcp 172.16.175.114:41780->52.154.154.117:6443: read: connection timed out
```
/retest

File: pkg/storage/azure/azure_test.go [hunk 1]
```diff
@@ -22,3 +37,137 @@ func TestGenerateAccountName(t *testing.T) {
...
+
+	listers := testBuilder.BuildListers()
+
+	authorizer := autorest.NullAuthorizer{}
+	sender := mocks.NewSender()
+	sender.AppendResponse(mocks.NewResponseWithContent(`{"nameAvailable":true}`))
+	sender.AppendResponse(mocks.NewResponseWithContent(`?`))
+	sender.AppendResponse(mocks.NewResponseWithContent(`{"name":"account"}`))
+	sender.AppendResponse(mocks.NewResponseWithContent(`{"keys":[{"value":"firstKey"}]}`))
+	sender.AppendResponse(mocks.NewResponseWithContent(`{"keys":[{"value":"firstKey"}]}`))
```
Comment (ricardomaraschini) [inline pkg/storage/azure/azure_test.go:92]: Can't we use `AppendAndRepeatResponse` here?

Comment (dmage) [reply pkg/storage/azure/azure_test.go:92]: No, we should eliminate the second call and I'd like to have the second call recorded explicitly

File: vendor/modules.txt [hunk 2]
```diff
@@ -24,6 +24,8 @@ github.com/Azure/go-autorest/autorest/azure/auth
 github.com/Azure/go-autorest/autorest/azure/cli
 # github.com/Azure/go-autorest/autorest/date v0.2.0
 github.com/Azure/go-autorest/autorest/date
+# github.com/Azure/go-autorest/autorest/mocks v0.3.0
```
[the remark this answers is not in the document]
Comment (dmage) [reply vendor/modules.txt:27]: It's part of the vendor commit

File: go.mod [hunk 3]
```diff
@@ -4,11 +4,12 @@ go 1.13
 
 require (
 	cloud.google.com/go v0.40.0
-	github.com/Azure/azure-pipeline-go v0.2.2 // indirect
```
[the remark this answers is not in the document]
Comment (dmage) [reply go.mod:7]: commits are reordered, go.mod is a part of the first commit

File: pkg/storage/azure/azure.go [hunk 4]
```diff
@@ -235,34 +249,69 @@ type driver struct {
 	Config     *imageregistryv1.ImageRegistryConfigStorageAzure
 	KubeConfig *rest.Config
```
[the remark this answers is not in the document]
Comment (dmage) [reply pkg/storage/azure/azure.go]: no, removed

File: pkg/storage/azure/azure.go [hunk 5]
```diff
@@ -235,34 +249,69 @@ type driver struct {
...
 	Listers    *regopclient.Listers
+
+	authorizer autorest.Authorizer
+	sender     autorest.Sender
+	httpSender pipeline.Factory
```
[the remark this answers is not in the document]
Comment (dmage) [reply pkg/storage/azure/azure.go:268]: added

**Comment by dmage** [comment]
AWS is not affected by this PR.
/test e2e-aws-image-registry
/test e2e-aws-upgrade

**Comment by dmage** [comment]
/hold
aws problems

**Pull request merged**